common/chains/client/rpc_client.go

```diff
@@ -0,0 +1,83 @@
+package client
+
+import (
+	"context"
+	"math/big"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+)
+
+type RPCClient[
+	CHAINID types.ID,
+	SEQ types.Sequence,
+	ADDR types.Hashable,
+	BLOCK any,
+	BLOCKHASH types.Hashable,
+	TX any,
+	TXHASH types.Hashable,
+	EVENT any,
+	EVENTOPS any, // event filter query options
+	TXRECEIPT any,
+	FEE feetypes.Fee,
+	HEAD types.Head[BLOCKHASH],
+	SUB types.Subscription,
+] interface {
+	Accounts[ADDR, SEQ]
+	Transactions[TX, TXHASH, TXRECEIPT]
+	Events[EVENT, EVENTOPS]
+	Blocks[BLOCK, BLOCKHASH]
+
+	BatchCallContext(ctx context.Context, b []any) error
+	CallContract(
+		ctx context.Context,
+		attempt interface{},
+		blockNumber *big.Int,
+	) (rpcErr []byte, extractErr error)
+	CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error
+	ChainID() (CHAINID, error)
+	CodeAt(ctx context.Context, account ADDR, blockNumber *big.Int) ([]byte, error)
+	ConfiguredChainID() CHAINID
+	EstimateGas(ctx context.Context, call any) (gas uint64, err error)
+	HeadByNumber(ctx context.Context, number *big.Int) (head HEAD, err error)
+	HeadByHash(ctx context.Context, hash BLOCKHASH) (head HEAD, err error)
+	LINKBalance(ctx context.Context, accountAddress ADDR, linkAddress ADDR) (*assets.Link, error)
+	PendingSequenceAt(ctx context.Context, addr ADDR) (SEQ, error)
```

Comment:
This could better be in Accounts interface?

Reply:
done!